test/specs/flat-map.js

```diff
@@ -239,4 +240,90 @@ describe('flatMap', () => {
       expect(result).to.flowErrors(c)
     })
   })
+
+  describe('overlapping with a concurrency limit that has maxed out', () => {
+    describe('and with a queue limit', () => {
+      it('not maxed out, should add to the queue', () => {
```

Comment:
All these "should add to the queue" tests are kinda silly because I can't really confirm that the queue has changed...